web/datapipeline/sap_systems_projector.go

```diff
@@ -79,31 +81,43 @@ func SAPSystemsProjector_SAPSystemsDiscoveryHandler(dataCollectedEvent *DataColl
 			instance.SystemReplicationStatus = parseReplicationStatus(i.SystemReplication)
 			addSAPControlData(&instance, i.SAPControl)
 
-			err := storeSAPInstance(db,
-				instance,
-				"id", "sid", "type", "features", "instance_number",
-				"system_replication", "system_replication_status",
-				"sap_hostname", "start_priority", "http_port", "https_port", "status",
-				"tenants", "db_host", "db_name")
+			instances = append(instances, instance)
+		}
+
+		err := storeSAPInstances(db,
+			instances,
+			"id", "sid", "type", "features", "instance_number",
+			"system_replication", "system_replication_status",
+			"sap_hostname", "start_priority", "http_port", "https_port", "status",
+			"tenants", "db_host", "db_name")
+		if err != nil {
+			return err
+		}
 
+		if len(instances) > 0 {
```

Comment:
What happens if we receive 0 instances suddenly?
We wouldn't remove anything, which looks wrong.
For instance, if at some point we uninstall the SAP system

Reply:
ah yes good call i totally forgot about this scenario i will add the case

Reply:
@fabriziosestito We could still have no SAP systems at all. This means that it wouldn't go within the 1st loop (`for _, s := range discoveredSAPSystems {`).

Reply:
@arbulu89 it should cover now both scenarios